### PR TITLE
chore(x/act): rename x/intent into x/act

### PR DESCRIPTION
**Pull request by pitasi**
Renaming:
x/intent -> x/act
Intent -> Rule

It should be clearer to understand as a concept.

While on it, I also wrapped Protobuf defs for x/act into a v1beta1 folder, for future updates.

### CODE REVIEW

File: proto/warden/warden/v1beta2/key.proto [hunk 1]
```diff
@@ -86,10 +86,10 @@ message Key {
   // generated it.
   bytes public_key = 5;
 
-  // ID of the intent that will need to be satisfied for using this key to sign
+  // ID of the Rule that will need to be satisfied for using this key to sign
   // data.
-  // If this is not set, the key will use the signing intent of the Space.
-  uint64 intent_id = 8;
+  // If this is not set, the key will use the signing Rule of the Space.
+  uint64 rule_id = 8;
```
Comment (artur-abliazimov) [inline proto/warden/warden/v1beta2/key.proto]: Must this be equal to 6 instead? Let's also review the `MsgNewSignatureRequest`.

File: proto/warden/warden/v1beta2/tx.proto [hunk 2]
```diff
@@ -159,7 +159,7 @@ message MsgNewKeyRequest {
   uint64 space_id = 2;
   uint64 keychain_id = 3;
   KeyType key_type = 4;
-  uint64 intent_id = 6;
+  uint64 rule_id = 6;
```
Comment (artur-abliazimov) [inline proto/warden/warden/v1beta2/tx.proto]: Must this be 5 instead?

File: warden/x/act/keeper/msg_server_update_rule.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	if rule.Creator != msg.Creator {
+		return nil, fmt.Errorf("account is not the creator of the rule")
+	}
+
+	rule.Expression = expr
+	rule.Name = msg.Name
```
Comment (artur-abliazimov) [inline warden/x/act/keeper/msg_server_update_rule.go:29]: Do we need to check the `Name` is not empty?

File: warden/x/act/types/v1beta1/types.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import (
+	"context"
+
+	proto "github.com/cosmos/gogoproto/proto"
+)
+
+// ActionHandler is a function that gets executed when an action is ready to be
+// fulfilled (i.e. it's rule has been satisfied).
+// The result of the action is stored in the Result field of the Action itself.
+type ActionHandler func(context.Context, Action) (proto.Message, error)
```
Comment (artur-abliazimov) [inline warden/x/act/types/v1beta1/types.go]: Must this be used explicitly? Can't find any usages.

File: warden/x/act/keeper/msg_server_update_rule.go (hunk 3 above)
Comment (pitasi) [reply warden/x/act/keeper/msg_server_update_rule.go:29]: mmm I won't add logic in this PR, but possibly we can add a check, yes

opened an issue: #439

File: warden/x/act/types/v1beta1/types.go (hunk 4 above)
Comment (pitasi) [reply warden/x/act/types/v1beta1/types.go]: I think this file was re-added automatically during a rebase. Will remove it again.

File: proto/warden/warden/v1beta2/tx.proto (hunk 2 above)
Comment (pitasi) [reply proto/warden/warden/v1beta2/tx.proto]: fixed

File: proto/warden/warden/v1beta2/key.proto (hunk 1 above)
Comment (pitasi) [reply proto/warden/warden/v1beta2/key.proto]: fixed

**Review by artur-abliazimov — APPROVED** [review]

**Pull request merged**